src/Host/Broker/Impl/Pipes/MessagePipe.cs

```diff
@@ -0,0 +1,173 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Threading.Tasks.Dataflow;
+using Microsoft.Extensions.Logging;
+using Microsoft.R.Host.Client;
+
+namespace Microsoft.R.Host.Broker.Pipes {
+    public class MessagePipe {
+        private readonly ILogger _logger;
+        private readonly BufferBlock<byte[]> _hostMessages = new BufferBlock<byte[]>();
```

Comment:
Sure.

---
In reply to: [74837927](https://github.com/Microsoft/RTVS/pull/2173#discussion_r74837927) [](ancestors = 74837927,74833910)